app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -185,8 +185,10 @@ public void setBlockingEnabled(boolean enabled) {
                 applyAppSettings();
             } else {
                 if (geckoSession != null) {
```

Comment:
I wonder why we do the `geckoSession != null` check here. Inside the block we do not use `geckoSession` and if it would be null then `geckoSession.getSettings()` at the beginning of the method would fail already.

Reply:
Yeah that makes sense